### PR TITLE
fix(css): enhance container-type property page

### LINKED ISSUE #29748: Please add clarification about affect on content-based sizing
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/container-type

### What specific section or headline is this issue about?

Values or other

### What information was incorrect, unhelpful, or incomplete?

This section:

> Note: to understand what happens when you apply layout, style, and size [contain](https://developer.mozilla.org/en-US/docs/Web/CSS/contain)ment to a box, see the contain property.

I don't think this is enough detail for people to understand the huge impact this property has on sizing, and the need to add an explicit size on all containers. 

### What did you expect to see?

This Stack Overflow question is a good example of the issue, and has a clear explanation of the answer: https://stackoverflow.com/questions/73975889/container-query-collapses-width-of-element/73980194#73980194

I think this MDN article should refer to this, becuase without it, users are left to discover this bomb for themselves. Having a link to the 'contain' property isn't enough, because that 'contain' article is very hard to understand and not clearly written to explain the potential issue with `container-type`. I think it should instead explain the sizing issue, then link to the 'contain' article for further reading. 

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

Thank you for considering this request. I'm sorry I don't feel confident enough with these issues to write a proposed update myself. 

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/container-type`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/container-type
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/container-type/index.md
* Last commit: https://github.com/mdn/content/commit/de76d381cb79a61813378963b66d98f68e74381c
* Document last modified: 2023-07-26T23:50:46.000Z

</details>

### PR DESCRIPTION
**Pull request by OnkarRuikar**
- fixes https://github.com/mdn/content/issues/29748

The PR
- adds more info about size containments in values section
- adds more details from the StackOverflow answer
- makes example complete and interactive

### CODE REVIEW

File: files/en-us/web/css/container-type/index.md [hunk 1]
```diff
@@ -81,11 +125,13 @@ Writing a container query via the {{Cssxref("@container")}} at-rule will apply s
 @container (min-width: 400px) {
   .card {
     display: grid;
-    grid-template-columns: 2fr 1fr;
+    grid-template-columns: 1fr 2fr;
   }
 }
 ```
 
+{{EmbedLiveSample('Establishing_inline_size_containment', '100%', 300)}}
```
Comment (chrisdavidmills) [inline files/en-us/web/css/container-type/index.md:133]: Nice work: the demo is instructive and easy to understand.

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]
@OnkarRuikar great work here, thanks! I have left a few comments for you to consider.

File: files/en-us/web/css/container-type/index.md [hunk 2]
```diff
@@ -32,16 +36,18 @@ container-type: unset;
   - : Establishes a query container for container size queries on both the inline and block axis in both the [inline and block](/en-US/docs/Web/CSS/CSS_logical_properties_and_values/Basic_concepts_of_logical_properties_and_values#block_and_inline_dimensions) dimensions.
```
Comment (OnkarRuikar) [inline files/en-us/web/css/container-type/index.md:36]:
```suggestion
  - : Establishes a query container for container size queries in both the [inline and block](/en-US/docs/Web/CSS/CSS_logical_properties_and_values/Basic_concepts_of_logical_properties_and_values#block_and_inline_dimensions) dimensions.
```

File: files/en-us/web/css/container-type/index.md [hunk 3]
```diff
@@ -32,16 +34,18 @@ container-type: unset;
...
```
[the remark this answers is not in the document]
Comment (OnkarRuikar) [reply files/en-us/web/css/container-type/index.md]: done in [this](https://github.com/mdn/content/pull/29852/commits/37327c848608294b68a3731e24b866c3ca9c9774) commit.

**Review by chrisdavidmills — APPROVED** [review]
LGTM; thanks @OnkarRuikar !

**Pull request merged**